tools/utils/utils.js

```diff
@@ -472,21 +479,21 @@ exports.isPathRelative = function (x) {
 //
 // This is talking about NPM/Cordova versions specifically, not Meteor versions.
 // It does not support the wrap number syntax.
-exports.ensureOnlyExactVersions = function (dependencies) {
+exports.ensureOnlyExactVersions = function (dependencies, {forNpm}) {
   _.each(dependencies, function (version, name) {
     // We want a given version of a smart package (package.js +
     // .npm/npm-shrinkwrap.json) to pin down its dependencies precisely, so we
     // don't want anything too vague. For now, we support semvers and urls that
     // name a specific commit by SHA.
-    if (! exports.isExactVersion(version)) {
+    if (! exports.isExactVersion(version, {forNpm})) {
       throw new Error(
         "Must declare exact version of dependency: " + name + '@' + version);
     }
   });
 };
-exports.isExactVersion = function (version) {
-  return semver.valid(version) || exports.isUrlWithSha(version)
-    || exports.isUrlWithFileScheme(version);
+exports.isExactVersion = function (version, {forNpm}) {
```

Comment:
Now that we no longer require a SHA in `isNpmUrl`, does `isExactVersion` still make sense? Or should we call it something like `isValidVersion`? (Coming to think of it, `isUrlWithFileScheme` never guaranteed an exact version.)